packages/web3-eth-contract/src/contract.ts

```diff
@@ -173,40 +176,45 @@ export class Contract<Abi extends ContractAbi>
 			throw new Web3ContractError('No constructor interface found.');
 		}
 
-		if (!deployOptions?.data) {
+		const data = format(
+			{ eth: 'bytes' },
+			deployOptions?.data ?? this.options.data,
+			DEFAULT_RETURN_FORMAT,
+		);
+
+		if (!data || data.trim() === '0x') {
 			throw new Web3ContractError('No data provided.');
 		}
-		const data = deployOptions?.data ?? this.options.data;
+
 		const args = deployOptions?.arguments ?? [];
-		validator.validate(['string'], args);
 
 		const contractOptions = { ...this.options, data };
 
 		return {
 			arguments: args,
-			send: async (options?: PayableCallOptions) => {
+			send: (
+				options?: PayableCallOptions,
```

Comment:
Its not eth_call so might be misleading to future developers, good to rename all its instances. 

```suggestion
				options?: PayableTxOptions,
```